client/src/api/datasource/LocalClient.ts

```diff
@@ -1,22 +1,99 @@
   }
-  update_meta(dataSource: string, filePath: string, meta: object): Promise<SigMFMetadata> {
-    throw new Error('Method not implemented.');
+
+  async getDataSourceMeta(account: string, container: string): Promise<SigMFMetadata[]> {
+    if (!store.getState().localClient?.files) {
+      return Promise.reject('No local directory found');
+    }
+    const localFiles: FileWithDirectoryAndFileHandle[] = store.getState().localClient.files;
+    let result: SigMFMetadata[] = [];
+    for (let i = 0; i < localFiles.length; i++) {
+      const file = localFiles[i];
+      if (file.name.split('.').pop() !== 'sigmf-meta') {
+        continue;
+      }
+      // check if there is a corresponding sigmf-data file
+      const dataFile = localFiles.find((f) => f.name === file.name.split('.')[0] + '.sigmf-data');
+      if (!dataFile) {
+        continue;
+      }
+      let metadata = await this.getMetaFromFile(file, dataFile, account, container);
+      result.push(metadata);
+    }
+    return result;
+  }
+  updateMeta(account: string, container: string, filePath: string, meta: object): Promise<SigMFMetadata> {
+    return Promise.reject('Not implemented');
   }
   list(): Promise<DataSource[]> {
-    throw new Error('Not implemented');
+    const localDirectory: FileWithDirectoryAndFileHandle[] = store.getState().localClient.files;
+    if (!localDirectory) {
+      return Promise.reject('No local directory found');
+    }
+    let directory = localDirectory[0];
+    return Promise.resolve([
+      {
+        name: directory.name,
+        account: 'local',
+        container: directory.webkitRelativePath.split('/')[0],
+        description: directory.name,
+      } as DataSource,
+    ]);
   }
-  get(dataSource: string): Promise<DataSource> {
-    throw new Error('Not implemented');
+  get(account: string, container: string): Promise<DataSource> {
+    return Promise.resolve({
+      name: container,
+      account: account,
+      container: container,
+      description: container,
+    } as DataSource);
   }
-  get_meta(dataSource: string, filePath: string): Promise<SigMFMetadata> {
-    throw new Error('Not implemented');
+  async getMeta(account: string, container: string, filePath: string): Promise<SigMFMetadata> {
+    const localDirectory: FileWithDirectoryAndFileHandle[] = store.getState().localClient.files;
+    if (!localDirectory) {
+      Promise.reject('No local directory found');
+    }
+    let metadataFile: FileWithDirectoryAndFileHandle | undefined = localDirectory.find((file) => {
+      return file.webkitRelativePath === filePath + '.sigmf-meta';
+    });
+    if (!metadataFile) {
+      return Promise.reject('No file found');
+    }
+    const dataFile = localDirectory.find((file) => {
+      return file.webkitRelativePath === filePath + '.sigmf-data';
+    });
+    if (!dataFile) {
+      return Promise.reject('No file found');
+    }
+    return this.getMetaFromFile(metadataFile, dataFile, account, container);
```

Comment:
I do not understand why the linter doesn't complain here, MAKE IT MAKE SENSE

Reply:
Yeha I thought aboput that the file kept growing btu changint i now would reuire to change the iomports of the changes that were done already in the other commit so I would say this si something that could be done after 👍

Reply:
It should not complain here I am glad ity doesnt we need to chack in the other times when it does complain for an else that is not neded.